### DISCUSSION TITLE
Custom style for all LXQt and Qt Applications

### DISCUSSION THREAD

**Discussion by hamzehnasajpour** [lxqt/lxqt#2073] [General]
I set these colors in `Appearance`:
![ScreenShot-2021-09-12_14-22-22](https://user-images.githubusercontent.com/5655083/132983154-e7135fb5-f851-49cb-b87a-3a9128544822.png)

In-fact I wanted to have a custom style with these:
1. Top Menu and Top bar: `black`
2. Content View: `white`
3. Left Panel: `black`

Like this screenshot:
![ScreenShot-2021-09-12_14-26-45](https://user-images.githubusercontent.com/5655083/132983273-1c7c8474-47fa-4f01-8cf1-4f7377df895e.png)

With above `palette` there are some issues in some applications, for example:
1. `lxqt-about`: `black` texts on `grey` background.
![ScreenShot-2021-09-12_14-29-07](https://user-images.githubusercontent.com/5655083/132983441-bbff7083-9eb6-4736-83fc-ef0dfca06348.png)

2. `pcmanfm-qt`: `grey` menu. While I expect to have a menu with `black` background.

![ScreenShot-2021-09-12_14-30-07](https://user-images.githubusercontent.com/5655083/132983443-120a0a86-5227-4519-9f25-9e59a82f8d0e.png)

Generally I want to have custom style, while it should be compatible for other LXQt/Qt applications. The point is I want to have top/left as `dark` and `content/view` as `white`. Can I achieve it via `lxqt-config-appearance` and `palette` settings? or I should implement my own `Qt Style plugin`? or any other idea.

A style like this [here](https://github.com/horst3180/arc-theme):
![687474703a2f2f692e696d6775722e636f6d2f4e43366471796c2e706e67](https://user-images.githubusercontent.com/5655083/132983545-329abeae-7f56-494d-ac2f-6f708af3e90c.png)

**Comment by tsujan** [comment]
> With above palette there are some issues in some applications

I think I emphasized it before; I say it here too:

*A palette whose window and view colors have a high contrast with each other will result in unreadable texts in some apps sooner or later.*

> While I expect to have a menu with black background.

First, those 8 colors aren't the whole palette. Qt makes the other colors based on them.

Second, You use Breeze, which adds its own colors. I quote [Qt doc](https://doc.qt.io/qt-5/qpalette.html):

*Some styles do not use the palette for all drawing...*

The default Qt style, which is called Fusion, uses palettes.

> Generally I want to have custom style...

You could use one of the available Qt widget styles or make your own. LXQt is compatible with all Qt widget styles.

**Comment by tsujan** [comment]
Last but not least, there's a fundamental difference between "style" and "theme".

A Qt style is a program (a Qt plugin) that may accept different themes. Some styles (like Fusion and Breeze) draw widgets always in the same way (rectangular buttons, identical shapes for check marks, etc.) but the colors they use can be set by palettes. Some other styles (like Kvantum) may be able to draw widgets very differently in their different themes. 

The palette section of LXQt Appearance Configuration is mostly for Fusion but it also affects Qt styles that accept palettes.
